src/django/api/views.py

```diff
@@ -2125,6 +2125,7 @@ def report(self, request, pk=None):
             reported_by_contributor=contributor,
             closure_state=request.data.get('closure_state'),
             reason_for_report=request.data.get('reason_for_report'))
+        facility_activity_report.full_clean()
```

Comment:
It looks like we need to go a bit further with this validation. When I tried POSTing "foo" as a status the API returned a full Django HTTP unhandled exception page.

<img width="992" alt="Screen Shot 2021-02-08 at 11 08 00 AM" src="https://user-images.githubusercontent.com/17363/107262769-306d2100-69fe-11eb-8d53-d1e889101921.png">

We have some existing code that handles a similar situation, but we will want to use a different exception.

https://github.com/open-apparel-registry/open-apparel-registry/blob/34ecf9d3e4a71235c4cd0fa3f8bbe71fd613e078/src/django/api/serializers.py#L68-L76

In summary, we have the catch the django.core.ValidationError that is raised by `full_clean` and raise an `api.exceptions.BadRequestException` instead. DRF will catch this and render a JSON response instead of returning HTML.

Reply:
Updated this in c65d115. Thanks for the assistance!